src/test/java/io/harness/cf/client/api/FFUseCaseTest.java

```diff
@@ -5,19 +5,35 @@
 import lombok.NonNull;
 import lombok.extern.slf4j.Slf4j;
 import org.testng.Assert;
+import org.testng.ITest;
 import org.testng.annotations.Test;
+import org.testng.annotations.BeforeMethod;
+
+import java.lang.reflect.Method;
 
 @Slf4j
-public class FFUseCaseTest {
+public class FFUseCaseTest implements ITest {
 
   private final TestCase testCase;
   private final Evaluator evaluator;
 
+  private ThreadLocal<String> testName = new ThreadLocal<>();
+
   public FFUseCaseTest(@NonNull final TestCase testCase, @NonNull final Evaluator evaluator) {
     this.testCase = testCase;
     this.evaluator = evaluator;
   }
 
+  @BeforeMethod
+  public void BeforeMethod(Method method, Object[] testData){
+    testName.set(testCase.getTestName());
+  }
+
+  @Override
+  public String getTestName() {
+    return testName.get();
```

Comment:
maybe you can use testcase member with some fields and return test name

Reply:
I've modified the BeforeMethod that sets the name to now be 
```java
 testName.set(testCase.getTestName() + "_with_target_" +testCase.getTargetIdentifier());
```